src/GraphQL/Utilities/Visitors/RemoveFederationTypesVisitor.cs

```diff
@@ -48,21 +51,50 @@ public static void Visit(ASTNode node)
     }
 
     /// <inheritdoc/>
-    protected override ValueTask VisitDocumentAsync(GraphQLDocument document, NullVisitorContext context)
+    protected override ValueTask VisitDocumentAsync(GraphQLDocument document, Context context)
     {
+        // assume the query type name is Query
+        context.QueryTypeName = "Query";
+
+        // scan for the schema definition first, to get the query type name
+        foreach (var definition in document.Definitions)
+        {
+            if (definition is GraphQLSchemaDefinition schemaDefinition)
+            {
+                foreach (var operationType in schemaDefinition.OperationTypes)
+                {
+                    if (operationType.Operation == OperationType.Query && operationType.Type != null) // operationType.Type should never be null
+                    {
+                        context.QueryTypeName = operationType.Type.Name;
+                        break;
+                    }
+                }
+                break;
+            }
+        }
+
+        // remove all federation directives and federation types
         document.Definitions.RemoveAll(node => node switch
         {
             GraphQLDirectiveDefinition directive => _federatedDirectives.Contains(directive.Name.StringValue),
             GraphQLTypeDefinition type => _federatedTypes.Contains(type.Name.StringValue),
             _ => false,
         });
+
         return base.VisitDocumentAsync(document, context);
     }
 
     /// <inheritdoc/>
-    protected override ValueTask VisitObjectTypeDefinitionAsync(GraphQLObjectTypeDefinition objectTypeDefinition, NullVisitorContext context)
+    protected override ValueTask VisitSchemaDefinitionAsync(GraphQLSchemaDefinition schemaDefinition, Context context)
+    {
+        schemaDefinition.Directives?.Items.RemoveAll(directive => directive.Name.Value == LINK_DIRECTIVE);
```

Comment:
This PR also removes the `@link` directive from the schema, which it did not before.